Polyhedron/demo/Polyhedron/Viewer.cpp

```diff
@@ -761,3 +871,148 @@ void Viewer::resizeGL(int w, int h)
     rendering_program.release();
 
 }
+
+QOpenGLShaderProgram* Viewer::getShaderProgram(int name) const
+{
+    // workaround constness issues in Qt
+    Viewer* viewer = const_cast<Viewer*>(this);
+
+    switch(name)
+    {
+    /// @TODO: factorize this code
+    case PROGRAM_WITH_LIGHT:
```

Comment:
@maxGimeno: The following lines are copy-pasted. It should not be too difficult to factorize the five cases in a single case. The names of the shaders could be stored in a constant array.